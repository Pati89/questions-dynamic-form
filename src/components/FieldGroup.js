import PropTypes from "prop-types";
import Input from "./Input";
import Select from "./Select";
import styles from "../styles/components/FieldGroup.module.css";

const FieldGroup = (props) => {
  const { fields, title, values, handleChange } = props;

  return (
    <div className={styles.fieldGroup} data-testid="field-group">
      <h2>{title}</h2>
      {fields?.map((item, index) =>
        item.type === "text" ? (
          <Input
            key={index}
            field={item}
            value={values[item.name]}
            onChange={handleChange}
          />
        ) : (
          <Select
            key={index}
            field={item}
            value={values[item.name]}
            onChange={handleChange}
          />
        )
      )}
    </div>
  );
};

FieldGroup.propTypes = {
  title: PropTypes.string,
  fields: PropTypes.array,
};

export default FieldGroup;
