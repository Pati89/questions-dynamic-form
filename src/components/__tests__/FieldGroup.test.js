import FieldGroup from "../FieldGroup";
import { render, screen } from "@testing-library/react";

const props = {
  values: {},
  title: "Test title",
  fields: [
    { name: "test1", label: "Test 1", type: "text" },
    { name: "test2", label: "Test 2", type: "text" },
    {
      name: "test3",
      label: "Test 3",
      type: "dropdown",
      options: ["Option 1", "Option 2"],
    },
  ],
};

describe("FieldGroup", () => {
  test("should render a FieldGroup", () => {
    render(<FieldGroup {...props} />);
    const fieldGroup = screen.getByTestId("field-group");
    expect(fieldGroup).toBeTruthy();
  });

  test("should render a title", () => {
    render(<FieldGroup {...props} />);
    const fieldGroup = screen.getByTestId("field-group");
    expect(fieldGroup).toHaveTextContent("Test title");
  });

  test("should render text fields", () => {
    render(<FieldGroup {...props} />);
    const input = screen.getAllByTestId("input");
    expect(input).toHaveLength(2);
  });

  test("should render select field", () => {
    render(<FieldGroup {...props} />);
    const select = screen.getAllByTestId("select");
    expect(select).toHaveLength(1);
  });

  test("should render select options", () => {
    render(<FieldGroup {...props} />);
    const selectOption = screen.getAllByTestId("select-option");
    expect(selectOption).toHaveLength(3);
  });

  test("should render a FieldGroup match snapshot", () => {
    render(<FieldGroup {...props} />);
    const fieldGroup = screen.getByTestId("field-group");
    expect(fieldGroup).toMatchSnapshot();
  });
});
