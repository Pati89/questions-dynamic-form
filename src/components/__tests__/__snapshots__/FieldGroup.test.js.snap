// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldGroup should render a FieldGroup match snapshot 1`] = `
<div
  class="fieldGroup"
  data-testid="field-group"
>
  <h2>
    Test title
  </h2>
  <div
    class="inputField"
    data-testid="input"
  >
    <label
      for="test1"
    >
      Test 1
    </label>
    <input
      data-testid="input-field"
      field="[object Object]"
      id="test1-input"
      name="test1"
      type="text"
      value=""
    />
  </div>
  <div
    class="inputField"
    data-testid="input"
  >
    <label
      for="test2"
    >
      Test 2
    </label>
    <input
      data-testid="input-field"
      field="[object Object]"
      id="test2-input"
      name="test2"
      type="text"
      value=""
    />
  </div>
  <div
    class="selectField"
    data-testid="select"
  >
    <label
      for="test3"
    >
      Test 3
    </label>
    <select
      data-testid="select-field"
      field="[object Object]"
      id="test3-select"
      name="test3"
      type="text"
    >
      <option
        data-testid="select-option"
        value=""
      >
        --Please choose an option--
      </option>
      <option
        data-testid="select-option"
        value="Option 1"
      >
        Option 1
      </option>
      <option
        data-testid="select-option"
        value="Option 2"
      >
        Option 2
      </option>
    </select>
  </div>
</div>
`;
